param.go

```diff
@@ -217,10 +217,19 @@ func (ps paramSingle) getValue(c containerStore) (reflect.Value, bool) {
 	return _noValue, false
 }
 
-// builds the parameter using decorators, if any. If there are no decorators associated
-// with this parameter, _noValue is returned.
+// builds the parameter using decorators in all scopes that affect the
+// current scope, if there are any. If there are multiple Scopes that decorates
+// this parameter, the closest one to the Scope that invoked this will be used.
+// If there are no decorators associated with this parameter, _noValue is returned.
 func (ps paramSingle) buildWithDecorators(c containerStore) (v reflect.Value, found bool, err error) {
-	decorators := c.getValueDecorators(ps.Name, ps.Type)
+	var decorators []decorator
+	var decoratingScope containerStore
```

Comment:
nit: group these
```suggestion
	var (
		decorators []decorator
		decoratingScope containerStore
	)
```

https://github.com/uber-go/guide/blob/master/style.md#group-similar-declarations